stl/inc/variant

```diff
@@ -1230,17 +1239,17 @@ public:
 
 private:
     template <size_t _Idx, class... _ArgTypes>
-    _Meta_at_c<variant, _Idx>& _Emplace_valueless(_ArgTypes&&... _Args) noexcept(
+    _CONSTEXPR20_DYNALLOC _Meta_at_c<variant, _Idx>& _Emplace_valueless(_ArgTypes&&... _Args) noexcept(
         is_nothrow_constructible_v<_Meta_at_c<variant, _Idx>, _ArgTypes...>) {
         // initialize alternative _Idx from _Args...
-        // pre: valueless_by_exception()
-        auto& _Obj = _Variant_raw_get<_Idx>(_Storage());
-        _Construct_in_place(_Obj, static_cast<_ArgTypes&&>(_Args)...);
+        _STL_INTERNAL_CHECK(valueless_by_exception());
+        _Construct_in_place(_Storage(), integral_constant<size_t, _Idx>{}, static_cast<_ArgTypes&&>(_Args)...);
         this->_Set_index(_Idx);
-        return _Obj;
+        return _Variant_raw_get<_Idx>(_Storage());

```

Comment:
This is a functional change on how we construct the element. 

I think it should not affect performance but please audit carefully

Reply:
I'm not super pleased about the quadratic code size, but I don't think we can do any better with our linear recursive union structure. (Refactoring `_Variant_storage` into some kind of tree structure would be amazingly cool, but just ever so slightly out of scope for this PR.)